### DISCUSSION TITLE
Did you ever figure out why the sql query was so slow

### DISCUSSION THREAD

**Discussion by Vilhelm-Ian** [mortii/anki-morphs#72] [General]
Recalc once used an sql query that took 8 minutes on my collection to recalc. Did you ever figure out why it was so slow

**Comment by mortii** [comment]
The old approach used `mw.col.db.executemany` which directly updates the db and _should_ be really fast. The new approach uses `mw.col.update_cards` which handles everything on the rust backend. I assume its better because rust is just way faster than python at everything .

**Reply by mortii** [reply]
Not sure why it was so slow for you and not me, maybe because my cpu is more powerful, the sql bottleneck is cpu.

**Reply by mortii** [reply]
maybe that is the reason actually, maybe it runs multithreaded in the rust backend or something

**Reply by mortii** [reply]
python sqlite only runs on one thread

**Comment by mortii** [comment]
was morphman slow for you? it uses the old approach

**Reply by Vilhelm-Ian** [reply]
It was slow but not that slow. I think it had something to do with the fact that the due date of a lot of cards was changed in your approach

**Comment by mortii** [comment]
I think the answer to your question is basically: rust > python
